reference/conanfile/attributes.rst

```diff
@@ -458,48 +504,38 @@ consistent implementation take into account these considerations:
 - A different behavior has ``self.options.option = None``, because
   ``assert self.options.option != None``.
 
-If you want to do a safe check of options values, you could use the ``get_safe()`` method:
-
-.. code-block:: python
-
-    def build(self):
-        # Will be None if doesn't exist
-        fpic = self.options.get_safe("fPIC")
-        # Will be the default version if the return is None
-        shared = self.options.get_safe("shared", default=False)
-
-The ``get_safe()`` method will return ``None`` if that option doesn't exist and there is no default value assigned.
 
 
 .. _conanfile_default_options:
 
 default_options
 ---------------
 
-As you have seen in the examples above, recipe's default options are declared as a dictionary with the initial desired value of the options.
-However, you can also specify default option values of the required dependencies:
-
-.. code-block:: python
-
-    class OtherPkg(ConanFile):
-        requires = "pkg/0.1@user/channel"
-        default_options = {"pkg:pkg_option": "value"}
-
-And it also works with default option values of conditional required dependencies:
+The attribute ``default_options`` has the purpose of defining the default values for the options
+if the consumer (consuming recipe, project, profile or the user through the command line) does
+not define them. This attribute should be defined as a python dictionary:
 
 .. code-block:: python
 
-    class OtherPkg(ConanFile):
-        default_options = {"pkg:pkg_option": "value"}
+    class MyPkg(ConanFile):
+        ...
+        options = {"shared": [True, False],
+                   "option1": ["value1", "value2"],
+                   "option2": "ANY"}
+        default_options = {"shared": True,
+                           "option1": "value1",
+                           "option2": 42}
 
-        def requirements(self):
-            if self.settings.os != "Windows":
-                self.requires("pkg/0.1@user/channel")
+        def build(self):
+            shared = "-DBUILD_SHARED_LIBS=ON" if self.options.shared else "-DBUILD_SHARED_LIBS=OFF"
```

Comment:
This BUILD_SHARED_LIBS is probably not a good example, as this is already done by ``CMake`` helper. Chose a generic ``cmake.definition`` based on ``option1`` or something like that.